controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -226,13 +230,18 @@ func (w *Workload) RemoveMachineFromKubeadmConfigMap(ctx context.Context, machin
 		return nil
 	}
 
+	return w.RemoveNodeFromKubeadmConfigMap(ctx, machine.Status.NodeRef.Name)
+}
+
+// RemoveNodeFromKubeadmConfigMap removes the entry for the node from the kubeadm configmap.
+func (w *Workload) RemoveNodeFromKubeadmConfigMap(ctx context.Context, name string) error {
```

Comment:
I'm not crazy about having two methods for doing the same thing, maybe can we just keep `RemoveNodeFromKubeadmConfigMap` and move the nil checks onto the current callers of `RemoveMachineFromKubeadmConfigMap`?

Reply:
I opened an issue for this. If you think it is something we need to do, feel free to reopen this: https://github.com/kubernetes-sigs/cluster-api/issues/2852